Lib/test/test_lltrace.py

```diff
@@ -1,31 +1,105 @@
-import os
+import dis
+import sys
 import textwrap
 import unittest
 
 from test.support import os_helper
 from test.support.script_helper import assert_python_ok
 
+def example():
+    x = []
+    for i in range(1):
+        x.append(i)
+    x = "this is"
+    y = "an example"
+    print(x, y)
 
+Py_DEBUG = hasattr(sys, 'gettotalrefcount')
+
+@unittest.skipUnless(Py_DEBUG, "lltrace requires Py_DEBUG")
 class TestLLTrace(unittest.TestCase):
 
+    def run_code(self, code):
+        code = textwrap.dedent(code).strip()
+        with open(os_helper.TESTFN, 'w', encoding='utf-8') as fd:
+            self.addCleanup(os_helper.unlink, os_helper.TESTFN)
+            fd.write(code)
+        status, stdout, stderr = assert_python_ok(os_helper.TESTFN)
+        self.assertEqual(stderr, b"")
+        self.assertEqual(status, 0)
+        return stdout.decode('utf-8')
+
+    def test_lltrace(self):
+        stdout = self.run_code("""
+            def dont_trace_1():
+                a = "a"
+                a = 10 * a
+            def trace_me():
+                for i in range(3):
+                    +i
+            def dont_trace_2():
+                x = 42
+                y = -x
+            dont_trace_1()
+            __ltrace__ = 1
+            trace_me()
+            del __ltrace__
+            dont_trace_2()
+        """)
+        self.assertIn("GET_ITER", stdout)
+        self.assertIn("FOR_ITER", stdout)
+        self.assertIn("UNARY_POSITIVE", stdout)
+        self.assertIn("POP_TOP", stdout)
+        self.assertNotIn("BINARY_OP", stdout)
+        self.assertNotIn("UNARY_NEGATIVE", stdout)
```

Comment:
Added printing on `-v`. I intentionally left the instruction offsets out of this test, since those are more a pain to fix when things change. If some opcode changed, hopefully only that one line would have to change here.